# Preventing microb to read partial cached content. It causes some problems on download stuff.
# Solved bug: 43765, broken xpinstaller functionality
--- mozilla/netwerk/protocol/http/src/nsHttpChannel.cpp.orig	2007-06-20 11:35:55.000000000 -0400
+++ mozilla/netwerk/protocol/http/src/nsHttpChannel.cpp	2007-06-20 11:37:02.000000000 -0400
@@ -765,20 +765,21 @@ nsHttpChannel::ProcessResponse()
             Cancel(NS_ERROR_NOT_RESUMABLE);
             rv = CallOnStartRequest();
             break;
         }
         // these can normally be cached
         rv = ProcessNormal();
         break;
     case 206:
-        if (mCachedContentIsPartial) // an internal byte range request...
-            rv = ProcessPartialContent();
-        else
-            rv = ProcessNormal();
+        // do not load partial cached content for nsICache::STORE_IN_MEMORY.
+        // if (mCachedContentIsPartial) // an internal byte range request...
+        //      rv = ProcessPartialContent();
+        // else
+        rv = ProcessNormal();
         break;
     case 300:
     case 301:
     case 302:
     case 307:
     case 303:
 #if 0
     case 305: // disabled as a security measure (see bug 187996).
